site/app/views/NavigationView.php

```diff
@@ -242,21 +249,216 @@ public function showGradeables($sections_to_list) {
 
                     $display_date = ($title == "FUTURE" || $title == "BETA") ? "<span style=\"font-size:smaller;\">(opens ".$g_data->getOpenDate()->format("m/d/y{$time}")."</span>)" : "<span style=\"font-size:smaller;\">(due ".$g_data->getDueDate()->format("m/d/y{$time}")."</span>)";
                     if ($title=="GRADED" || $title=="ITEMS BEING GRADED") { $display_date = ""; }
-                    $button_text = "{$title_to_prefix[$title]} {$submission_status["SUBMITTED"]} {$submission_status["AUTOGRADE"]} {$display_date}";
+                    if ($g_data->getActiveVersion() >= 1 && $title == "OPEN") { //if the user submitted something on time
+                        $button_text = "RESUBMIT {$submission_status["SUBMITTED"]} {$submission_status["AUTOGRADE"]} {$display_date}";
+                    }
+                    else if($g_data->getActiveVersion() >= 1 && $title_save == "CLOSED") { //if the user submitted something past time
+                        $button_text = "LATE RESUBMIT {$submission_status["SUBMITTED"]} {$submission_status["AUTOGRADE"]} {$display_date}";
+                    }
+                    else if(($title_save == "GRADED" || $title_save == "ITEMS BEING GRADED") && $g_data->getActiveVersion() < 1) {
+                    	//to change the text to overdue submission if nothing was submitted on time
+                        $button_text = "OVERDUE SUBMISSION {$submission_status["SUBMITTED"]} {$submission_status["AUTOGRADE"]} {$display_date}";
+                    } //when there is no TA grade and due date passed
+                    else if($title_save == "GRADED" && $g_data->useTAGrading() && !$g_data->beenTAgraded()) { 
+                        $button_text = "TA GRADE NOT AVAILABLE {$submission_status["SUBMITTED"]} 
+                        	{$submission_status["AUTOGRADE"]} {$display_date}";
+                        $title_to_button_type_submission['GRADED'] = "btn-default";
+                    }
+                    else {
+                    	$button_text = "{$title_to_prefix[$title]} {$submission_status["SUBMITTED"]} {$submission_status["AUTOGRADE"]} {$display_date}";
+                    }
                     if ($g_data->hasConfig()) {
-                        $gradeable_open_range = <<<HTML
+                        //calculate the point percentage
+                    	if ($g_data->getTotalNonHiddenNonExtraCreditPoints() == 0) {
+                    		$points_percent = 0;
+                    	}
+                    	else {
+                    		$points_percent = $g_data->getGradedNonHiddenPoints() / $g_data->getTotalNonHiddenNonExtraCreditPoints();
+                    	}                    	
+						$points_percent = $points_percent * 100;
+						if ($points_percent > 100) { $points_percent = 100; }
+                        //prints out the button if $points_percent >= 50% or if there is no autograding points
```

Comment:
This comment (and the one on the else) don't really add any sort of clarification that isn't gleamed from just reading the if statement conditionals.